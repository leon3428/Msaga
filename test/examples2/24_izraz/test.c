int main(void) {
    int x = ++(10 * 11 + 69 - 420 & 10 | 3 ^ 4);

    return 0;
}