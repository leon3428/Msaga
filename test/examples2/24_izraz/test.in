<prijevodna_jedinica>
 <vanjska_deklaracija>
  <definicija_funkcije>
   <ime_tipa>
    <specifikator_tipa>
     KR_INT 1 int
   IDN 1 main
   L_ZAGRADA 1 (
   KR_VOID 1 void
   D_ZAGRADA 1 )
   <slozena_naredba>
    L_VIT_ZAGRADA 1 {
    <lista_deklaracija>
     <deklaracija>
      <ime_tipa>
       <specifikator_tipa>
        KR_INT 2 int
      <lista_init_deklaratora>
       <init_deklarator>
        <izravni_deklarator>
         IDN 2 x
        OP_PRIDRUZI 2 =
        <inicijalizator>
         <izraz_pridruzivanja>
          <log_ili_izraz>
           <log_i_izraz>
            <bin_ili_izraz>
             <bin_xili_izraz>
              <bin_i_izraz>
               <jednakosni_izraz>
                <odnosni_izraz>
                 <aditivni_izraz>
                  <multiplikativni_izraz>
                   <cast_izraz>
                    <unarni_izraz>
                     OP_INC 2 ++
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        L_ZAGRADA 2 (
                        <izraz>
                         <izraz_pridruzivanja>
                          <log_ili_izraz>
                           <log_i_izraz>
                            <bin_ili_izraz>
                             <bin_ili_izraz>
                              <bin_xili_izraz>
                               <bin_i_izraz>
                                <bin_i_izraz>
                                 <jednakosni_izraz>
                                  <odnosni_izraz>
                                   <aditivni_izraz>
                                    <aditivni_izraz>
                                     <aditivni_izraz>
                                      <multiplikativni_izraz>
                                       <multiplikativni_izraz>
                                        <cast_izraz>
                                         <unarni_izraz>
                                          <postfiks_izraz>
                                           <primarni_izraz>
                                            BROJ 2 10
                                       OP_PUTA 2 *
                                       <cast_izraz>
                                        <unarni_izraz>
                                         <postfiks_izraz>
                                          <primarni_izraz>
                                           BROJ 2 11
                                     PLUS 2 +
                                     <multiplikativni_izraz>
                                      <cast_izraz>
                                       <unarni_izraz>
                                        <postfiks_izraz>
                                         <primarni_izraz>
                                          BROJ 2 69
                                    MINUS 2 -
                                    <multiplikativni_izraz>
                                     <cast_izraz>
                                      <unarni_izraz>
                                       <postfiks_izraz>
                                        <primarni_izraz>
                                         BROJ 2 420
                                OP_BIN_I 2 &
                                <jednakosni_izraz>
                                 <odnosni_izraz>
                                  <aditivni_izraz>
                                   <multiplikativni_izraz>
                                    <cast_izraz>
                                     <unarni_izraz>
                                      <postfiks_izraz>
                                       <primarni_izraz>
                                        BROJ 2 10
                             OP_BIN_ILI 2 |
                             <bin_xili_izraz>
                              <bin_xili_izraz>
                               <bin_i_izraz>
                                <jednakosni_izraz>
                                 <odnosni_izraz>
                                  <aditivni_izraz>
                                   <multiplikativni_izraz>
                                    <cast_izraz>
                                     <unarni_izraz>
                                      <postfiks_izraz>
                                       <primarni_izraz>
                                        BROJ 2 3
                              OP_BIN_XILI 2 ^
                              <bin_i_izraz>
                               <jednakosni_izraz>
                                <odnosni_izraz>
                                 <aditivni_izraz>
                                  <multiplikativni_izraz>
                                   <cast_izraz>
                                    <unarni_izraz>
                                     <postfiks_izraz>
                                      <primarni_izraz>
                                       BROJ 2 4
                        D_ZAGRADA 2 )
      TOCKAZAREZ 2 ;
    <lista_naredbi>
     <naredba>
      <naredba_skoka>
       KR_RETURN 4 return
       <izraz>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      BROJ 4 0
       TOCKAZAREZ 4 ;
    D_VIT_ZAGRADA 5 }
