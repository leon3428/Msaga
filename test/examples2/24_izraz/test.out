<unarni_izraz> ::= OP_INC(2,++) <unarni_izraz>
